editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,350 @@
+  private Feature selectedFeature = null;
+  private boolean selectedFeatureIsPolyline = false;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add JavaFX scene to stage
+      stage.setTitle("Edit Features With Feature-linked Annotation Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create the map with a light gray canvas basemap centered on Loudoun, Virginia
+      ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS_VECTOR, 39.0204, -77.4159, 18);
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // add the map view to stack pane
+      stackPane.getChildren().addAll(mapView);
+      
+      // create and load the geodatabase
+      File geodatabaseFile = new File(System.getProperty("data.dir"),
+        "./samples-data/loudon/loudoun_anno.geodatabase");
+      Geodatabase geodatabase = new Geodatabase(geodatabaseFile.getAbsolutePath());
+
+      geodatabase.addDoneLoadingListener(() -> {
+        if (geodatabase.getLoadStatus() == LoadStatus.LOADED) {
+          // create feature layers from tables in the geodatabase
+          addressPointFeatureLayer = new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("Loudoun_Address_Points_1"));
+          parcelLinesFeatureLayer = new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("ParcelLines_1"));
+          // create annotation layers from tables in the geodatabase
+          addressPointsAnnotationLayer = new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("Loudoun_Address_PointsAnno_1"));
+          parcelLinesAnnotationLayer = new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("ParcelLinesAnno_1"));
+
+          // add the feature layers to the map
+          map.getOperationalLayers().add(addressPointFeatureLayer);
+          map.getOperationalLayers().add(parcelLinesFeatureLayer);
+
+          // add the annotation layers to the map
+          map.getOperationalLayers().add(addressPointsAnnotationLayer);
+          map.getOperationalLayers().add(parcelLinesAnnotationLayer);
+        } else {
+          // show alert if geodatabase fails to load
+          new Alert(Alert.AlertType.ERROR, "Error loading Geodatabase.").show();
+        }
+      });
+      geodatabase.loadAsync();
+
+      // set on click behaviour
+      mapView.setOnMouseClicked(event -> {
+        // check that the primary mouse button was clicked
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // create a point where the user clicked
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+          // call select or move method to move to the point
+          selectOrMove(screenPoint);
```

Comment:
it's common with the Android implementations to have single use methods, but with Java SE samples we don't do this as often. The logic in this method isn't too verbose, so let's just remove the method and insert the code from that method directly here instead. 
```suggestion
    // if a feature hasn't been selected, select the feature
    if (selectedFeature == null) {
      identifyFeature(screenPoint);
    } else {
      // convert the screen point to a map point
      Point mapPoint = mapView.screenToLocation(screenPoint);
      // if the feature is a polyline, move the polyline
      if (selectedFeatureIsPolyline) {
        movePolylineVertex(mapPoint);
      } else {
        // if the feature is a point, move the point
        movePoint(mapPoint);
      }
    }```